pkg/apis/traceflow/v1/types.go

```diff
@@ -0,0 +1,158 @@
+	Service   string `json:"service,omitempty"`   // Service is the destination service, exlusive with destination pod.
+	IP        string `json:"IP,omitempty"`        // IP is the destination IP.
+}
+
+// IPHeader describes spec of an IPv4 header. IPv6 not supported yet.
+type IPHeader struct {
+	SrcIP    string `json:"srcIP,omitempty"`    // SrcIP is the source IP.
+	Protocol int32  `json:"protocol,omitempty"` // Protocol is the IP protocol.
+	TTL      int32  `json:"ttl,omitempty"`      // TTL is the IP TTL.
+	Flags    int32  `json:"flags,omitempty"`    // Flags is the flags for IP.
+}
+
+// TransportHeader describes spec of a TransportHeader.
+type TransportHeader struct {
+	ICMPEchoRequestHeader `json:"icmpEchoRequestHeader"`
+	UDPHeader             `json:"udpHeader"`
+	TCPHeader             `json:"tcpHeader"`
+}
+
+// ICMPEchoRequestHeader describes spec of an ICMP echo request header.
+type ICMPEchoRequestHeader struct {
+	ID       int32 `json:"id,omitempty"`       // ID is the ICMPEchoRequestHeader ID.
+	Sequence int32 `json:"sequence,omitempty"` // Sequence is the ICMPEchoRequestHeader sequence.
+}
+
+// UDPHeader describes spec of a UDP header.
+type UDPHeader struct {
+	SrcPort int32 `json:"srcPort,omitempty"` // SrcPort is the source port.
+	DstPort int32 `json:"dstPort,omitempty"` // DstPort is the destination port.
+}
+
+// TCPHeader describes spec of a TCP header.
+type TCPHeader struct {
+	SrcPort int32 `json:"srcPort,omitempty"` // SrcPort is the source port.
+	DstPort int32 `json:"dstPort,omitempty"` // DstPort is the destination port.
+	Flags   int32 `json:"flags,omitempty"`   // Flags are flags in the header.
+}
+
+// Packet includes payload info and headers info.
+type Packet struct {
+	PayloadString string `json:"payloadString,omitempty"` // PayloadString is the content in payload.
+	PayloadSize   int32  `json:"payloadSize,omitempty"`   // PayloadSize is the size of payload.
+
+	IPHeader        `json:"ipHeader"`
+	TransportHeader `json:"transportHeader"`
+}
+
+// Status describes current status of the traceflow.
+type Status struct {
+	Phase        Phase         `json:"phase,omitempty"`        // Phase is the Traceflow phase.
+	CrossNodeTag uint8         `json:"crossNodeTag,omitempty"` // CrossNodeTag is an ID encoded into packet traceflow header to identify inter-node packet.
+	SenderNode   []Observation `json:"senderNode,omitempty"`   // SenderNode is observation from sender node.
+	ReceiverNode []Observation `json:"receiverNode,omitempty"` // ReceiverNode is observation from receiver node.
+}
+
+// Observation describes those from sender nodes or receiver nodes.
+type Observation struct {
+	Component       Component  `json:"component,omitempty"`       // Component is the observation component.
+	Action          Action     `json:"action,omitempty"`          // Action is the action to the observation.
+	Node            string     `json:"node,omitempty"`            // Node is the node of the observation.
```

Comment:
If we want to use single slice instead of multiple slices for different Nodes, please add below property:
NodeRole string     `json:"nodeRole,omitempty"`